### ISSUE TITLE
Exclude option is no longer working

### ISSUE THREAD

**Issue by gone-for-coding** [ansible/ansible-lint#3508]
##### Summary

The option `--exclude` and in the configuration file `exclude_paths` is no longer working.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint --version
ansible-lint 6.16.2 using ansible-core:2.15.0 ruamel-yaml:0.17.28 ruamel-yaml-clib:0.2.7
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

Create an empty folder. Put in a minimal playbook like this one (I have one named good.yml with less issues as well):

bad.yml
```yaml
---
- hosts: localhost
  tasks:

    - name: Install Active Directory PowerShell feature
      ansible.windows.win_feature:
        name: RSAT-AD-PowerShell
        include_sub_features: no
        state: present
```

```console (paste below)
root@21f7bcb4b742:/code# ansible-lint --exclude bad.yml -vvvv
DEBUG    Logging initialized to level 10
INFO     Identified / as project root due file system root.
DEBUG    Options: Options(cache_dir=PosixPath('/root/.cache/ansible-compat/e3b0c4'), colored=True, configured=True, cwd=PosixPath('/code'), display_relative_path=True, exclude_paths=['bad.yml'], format=None, lintables=[], list_rules=False, list_tags=False, write_list=None, parseable=False, quiet=0, rulesdirs=[PosixPath('/usr/local/lib/python3.11/site-packages/ansiblelint/rules')], skip_list=[], tags=[], verbosity=4, warn_list=['experimental', 'jinja'], mock_filters=[], mock_modules=[], mock_roles=[], loop_var_prefix=None, only_builtins_allow_collections=[], only_builtins_allow_modules=[], var_naming_pattern=None, offline=None, project_dir='..', extra_vars=None, enable_list=[], skip_action_validation=True, strict=False, rules={}, profile=None, task_name_prefix='{stem} | ', sarif_file=None, config_file=None, generate_ignore=False, rulesdir=[], cache_dir_lock=<filelock._unix.UnixFileLock object at 0x7fec01e5a410>, use_default_rules=False, version=False, list_profiles=False, ignore_file=None)
DEBUG    CWD: /code
DEBUG    Effective yamllint rules used: {'anchors': {'level': 'error', 'forbid-undeclared-aliases': True, 'forbid-duplicated-anchors': False, 'forbid-unused-anchors': False}, 'braces': {'level': 'error', 'forbid': False, 'min-spaces-inside': 0, 'max-spaces-inside': 1, 'min-spaces-inside-empty': -1, 'max-spaces-inside-empty': -1}, 'brackets': {'level': 'error', 'forbid': False, 'min-spaces-inside': 0, 'max-spaces-inside': 0, 'min-spaces-inside-empty': -1, 'max-spaces-inside-empty': -1}, 'colons': {'level': 'error', 'max-spaces-before': 0, 'max-spaces-after': 1}, 'commas': {'level': 'error', 'max-spaces-before': 0, 'min-spaces-after': 1, 'max-spaces-after': 1}, 'comments': {'level': 'warning', 'require-starting-space': True, 'ignore-shebangs': True, 'min-spaces-from-content': 1}, 'comments-indentation': False, 'document-end': False, 'document-start': False, 'empty-lines': {'level': 'error', 'max': 2, 'max-start': 0, 'max-end': 0}, 'empty-values': False, 'float-values': False, 'hyphens': {'level': 'error', 'max-spaces-after': 1}, 'indentation': {'level': 'error', 'spaces': 'consistent', 'indent-sequences': True, 'check-multi-line-strings': False}, 'key-duplicates': {'level': 'error'}, 'key-ordering': False, 'line-length': {'level': 'error', 'max': 160, 'allow-non-breakable-words': True, 'allow-non-breakable-inline-mappings': False}, 'new-line-at-end-of-file': {'level': 'error'}, 'new-lines': {'level': 'error', 'type': 'unix'}, 'octal-values': {'forbid-implicit-octal': True, 'forbid-explicit-octal': True, 'level': 'error'}, 'quoted-strings': False, 'trailing-spaces': {'level': 'error'}, 'truthy': {'level': 'warning', 'allowed-values': ['true', 'false'], 'check-keys': True}}
INFO     Set ANSIBLE_LIBRARY=/root/.cache/ansible-compat/5694d0/modules:/root/.ansible/plugins/modules:/usr/share/ansible/plugins/modules
INFO     Set ANSIBLE_COLLECTIONS_PATH=/root/.cache/ansible-compat/5694d0/collections:/root/.ansible/collections:/usr/share/ansible/collections
INFO     Set ANSIBLE_ROLES_PATH=/root/.cache/ansible-compat/5694d0/roles:/root/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles
INFO     Looking up for files, excluding bad.yml ...
INFO     Executing syntax check on playbook good.yml (0.62s)
INFO     Executing syntax check on playbook bad.yml (0.63s)
DEBUG    Examining good.yml of type playbook
DEBUG    Examining bad.yml of type playbook
WARNING  Listing 3 violation(s) that are fatal
name[play]: All plays should be named.
bad.yml:2

yaml[truthy]: Truthy value should be one of [false, true]
bad.yml:8

name[casing]: All names should start with an uppercase letter.
good.yml:2

DEBUG    Attempting to release lock 140651620836368 on /root/.cache/ansible-compat/e3b0c4/.lock
DEBUG    Lock 140651620836368 released on /root/.cache/ansible-compat/e3b0c4/.lock
Read documentation for instructions on how to ignore specific rule violations.

DEBUG    Determined rule-profile order: {'internal-error': (0, 'min'), 'load-failure': (1, 'min'), 'parser-error': (2, 'min'), 'syntax-check': (3, 'min'), 'command-instead-of-module': (4, 'basic'), 'command-instead-of-shell': (5, 'basic'), 'deprecated-bare-vars': (6, 'basic'), 'deprecated-local-action': (7, 'basic'), 'deprecated-module': (8, 'basic'), 'inline-env-var': (9, 'basic'), 'key-order': (10, 'basic'), 'literal-compare': (11, 'basic'), 'jinja': (12, 'basic'), 'no-free-form': (13, 'basic'), 'no-jinja-when': (14, 'basic'), 'no-tabs': (15, 'basic'), 'partial-become': (16, 'basic'), 'playbook-extension': (17, 'basic'), 'role-name': (18, 'basic'), 'schema': (19, 'basic'), 'name': (20, 'basic'), 'var-naming': (21, 'basic'), 'yaml': (22, 'basic'), 'name': (23, 'moderate'), 'name': (24, 'moderate'), 'name': (25, 'moderate'), 'spell-var-name': (26, 'moderate'), 'avoid-implicit': (27, 'safety'), 'latest': (28, 'safety'), 'package-latest': (29, 'safety'), 'risky-file-permissions': (30, 'safety'), 'risky-octal': (31, 'safety'), 'risky-shell-pipe': (32, 'safety'), 'galaxy': (33, 'shared'), 'ignore-errors': (34, 'shared'), 'layout': (35, 'shared'), 'meta-incorrect': (36, 'shared'), 'meta-no-tags': (37, 'shared'), 'meta-video-links': (38, 'shared'), 'meta-version': (39, 'shared'), 'meta-runtime': (40, 'shared'), 'no-changed-when': (41, 'shared'), 'no-changelog': (42, 'shared'), 'no-handler': (43, 'shared'), 'no-relative-paths': (44, 'shared'), 'max-block-depth': (45, 'shared'), 'max-tasks': (46, 'shared'), 'unsafe-loop': (47, 'shared'), 'avoid-dot-notation': (48, 'production'), 'sanity': (49, 'production'), 'fqcn': (50, 'production'), 'import-task-no-when': (51, 'production'), 'meta-no-dependencies': (52, 'production'), 'single-entry-point': (53, 'production'), 'use-loop': (54, 'production')}
              Rule Violation Summary
 count tag          profile  rule associated tags
     1 name[play]   basic    idiom
     1 yaml[truthy] basic    formatting, yaml
     1 name[casing] moderate idiom

Failed after min profile: 3 failure(s), 0 warning(s) on 2 files.
```

##### Desired Behavior

Restore behaviour which works in 6.14.6 for example:

```console
root@71929bca4184:/code# ansible-lint --version
ansible-lint 6.14.6 using ansible 2.14.6
A new release of ansible-lint is available: 6.14.6 → 6.16.2
root@71929bca4184:/code# ansible-lint --exclude bad.yml -vvv
DEBUG    Logging initialized to level 10
DEBUG    Options: Namespace(cache_dir='/root/.cache/ansible-compat/5694d0', colored=True, configured=True, cwd=PosixPath('/code'), display_relative_path=True, exclude_paths=[PosixPath('/code/bad.yml')], format=None, lintables=[], list_rules=False, list_tags=False, write_list=None, parseable=False, quiet=0, rulesdirs=['/usr/local/lib/python3.11/site-packages/ansiblelint/rules'], skip_list=[], tags=[], verbosity=3, warn_list=['experimental', 'jinja[spacing]'], kinds=[{'jinja2': '**/*.j2'}, {'jinja2': '**/*.j2.*'}, {'yaml': '.github/**/*.{yaml,yml}'}, {'text': '**/templates/**/*.*'}, {'execution-environment': '**/execution-environment.yml'}, {'ansible-lint-config': '**/.ansible-lint'}, {'ansible-lint-config': '**/.config/ansible-lint.yml'}, {'ansible-navigator-config': '**/ansible-navigator.{yaml,yml}'}, {'inventory': '**/inventory/**.{yaml,yml}'}, {'requirements': '**/meta/requirements.{yaml,yml}'}, {'galaxy': '**/galaxy.yml'}, {'reno': '**/releasenotes/*/*.{yaml,yml}'}, {'tasks': '**/tasks/**/*.{yaml,yml}'}, {'rulebook': '**/rulebooks/*.{yml,yaml'}, {'playbook': '**/playbooks/*.{yml,yaml}'}, {'playbook': '**/*playbook*.{yml,yaml}'}, {'role': '**/roles/*/'}, {'handlers': '**/handlers/*.{yaml,yml}'}, {'vars': '**/{host_vars,group_vars,vars,defaults}/**/*.{yaml,yml}'}, {'test-meta': '**/tests/integration/targets/*/meta/main.{yaml,yml}'}, {'meta': '**/meta/main.{yaml,yml}'}, {'meta-runtime': '**/meta/runtime.{yaml,yml}'}, {'role-arg-spec': '**/meta/argument_specs.{yaml,yml}'}, {'yaml': '.config/molecule/config.{yaml,yml}'}, {'requirements': '**/molecule/*/{collections,requirements}.{yaml,yml}'}, {'yaml': '**/molecule/*/{base,molecule}.{yaml,yml}'}, {'requirements': '**/requirements.{yaml,yml}'}, {'playbook': '**/molecule/*/*.{yaml,yml}'}, {'yaml': '**/{.ansible-lint,.yamllint}'}, {'changelog': '**/changelogs/changelog.yaml'}, {'yaml': '**/*.{yaml,yml}'}, {'yaml': '**/.*.{yaml,yml}'}, {'sanity-ignore-file': '**/tests/sanity/ignore-*.txt'}], mock_filters=[], mock_modules=[], mock_roles=[], loop_var_prefix=None, only_builtins_allow_collections=[], only_builtins_allow_modules=[], var_naming_pattern=None, offline=None, project_dir='.', extra_vars=None, enable_list=[], skip_action_validation=True, strict=False, rules={}, profile=None, task_name_prefix='{stem} | ', sarif_file=None, list_profiles=False, progressive=False, rulesdir=[], use_default_rules=False, generate_ignore=False, config_file=None, ignore_file=None, version=False, cache_dir_lock=<filelock._unix.UnixFileLock object at 0x7f806800f210>)
DEBUG    /code
DEBUG    Checking for updated schemas...
DEBUG    Refreshing ansible-lint-config schema ...
INFO     Schema ansible-lint-config was updated
DEBUG    Refreshing ansible-navigator-config schema ...
INFO     Schema ansible-navigator-config was updated
DEBUG    Refreshing changelog schema ...
INFO     Schema changelog was updated
DEBUG    Refreshing execution-environment schema ...
INFO     Schema execution-environment was updated
DEBUG    Refreshing galaxy schema ...
DEBUG    Schema https://raw.githubusercontent.com/ansible/ansible-lint/main/src/ansiblelint/schemas/galaxy.json is not modified
DEBUG    Refreshing inventory schema ...
DEBUG    Schema https://raw.githubusercontent.com/ansible/ansible-lint/main/src/ansiblelint/schemas/inventory.json is not modified
DEBUG    Refreshing meta schema ...
INFO     Schema meta was updated
DEBUG    Refreshing meta-runtime schema ...
DEBUG    Schema https://raw.githubusercontent.com/ansible/ansible-lint/main/src/ansiblelint/schemas/meta-runtime.json is not modified
DEBUG    Refreshing molecule schema ...
DEBUG    Schema https://raw.githubusercontent.com/ansible/ansible-lint/main/src/ansiblelint/schemas/molecule.json is not modified
DEBUG    Refreshing playbook schema ...
INFO     Schema playbook was updated
DEBUG    Refreshing requirements schema ...
DEBUG    Schema https://raw.githubusercontent.com/ansible/ansible-lint/main/src/ansiblelint/schemas/requirements.json is not modified
DEBUG    Refreshing role-arg-spec schema ...
DEBUG    Schema https://raw.githubusercontent.com/ansible/ansible-lint/main/src/ansiblelint/schemas/role-arg-spec.json is not modified
DEBUG    Refreshing rulebook schema ...
INFO     Schema rulebook was updated
DEBUG    Refreshing tasks schema ...
INFO     Schema tasks was updated
DEBUG    Refreshing vars schema ...
DEBUG    Schema https://raw.githubusercontent.com/ansible/ansible-lint/main/src/ansiblelint/schemas/vars.json is not modified
DEBUG    Effective yamllint rules used: {'anchors': {'level': 'error', 'forbid-undeclared-aliases': True, 'forbid-duplicated-anchors': False, 'forbid-unused-anchors': False}, 'braces': {'level': 'error', 'forbid': False, 'min-spaces-inside': 0, 'max-spaces-inside': 1, 'min-spaces-inside-empty': -1, 'max-spaces-inside-empty': -1}, 'brackets': {'level': 'error', 'forbid': False, 'min-spaces-inside': 0, 'max-spaces-inside': 0, 'min-spaces-inside-empty': -1, 'max-spaces-inside-empty': -1}, 'colons': {'level': 'error', 'max-spaces-before': 0, 'max-spaces-after': 1}, 'commas': {'level': 'error', 'max-spaces-before': 0, 'min-spaces-after': 1, 'max-spaces-after': 1}, 'comments': {'level': 'warning', 'require-starting-space': True, 'ignore-shebangs': True, 'min-spaces-from-content': 1}, 'comments-indentation': False, 'document-end': False, 'document-start': False, 'empty-lines': {'level': 'error', 'max': 2, 'max-start': 0, 'max-end': 0}, 'empty-values': False, 'float-values': False, 'hyphens': {'level': 'error', 'max-spaces-after': 1}, 'indentation': {'level': 'error', 'spaces': 'consistent', 'indent-sequences': True, 'check-multi-line-strings': False}, 'key-duplicates': {'level': 'error'}, 'key-ordering': False, 'line-length': {'level': 'error', 'max': 160, 'allow-non-breakable-words': True, 'allow-non-breakable-inline-mappings': False}, 'new-line-at-end-of-file': {'level': 'error'}, 'new-lines': {'level': 'error', 'type': 'unix'}, 'octal-values': {'forbid-implicit-octal': True, 'forbid-explicit-octal': True, 'level': 'error'}, 'quoted-strings': False, 'trailing-spaces': {'level': 'error'}, 'truthy': {'level': 'warning', 'allowed-values': ['true', 'false'], 'check-keys': True}}
INFO     Set ANSIBLE_LIBRARY=/root/.cache/ansible-compat/5694d0/modules:/root/.ansible/plugins/modules:/usr/share/ansible/plugins/modules
INFO     Set ANSIBLE_COLLECTIONS_PATH=/root/.cache/ansible-compat/5694d0/collections:/root/.ansible/collections:/usr/share/ansible/collections
INFO     Set ANSIBLE_ROLES_PATH=/root/.cache/ansible-compat/5694d0/roles:/root/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles
INFO     Looking up for files, excluding /code/bad.yml ...
DEBUG    Ignored bad.yml due to: File /code/bad.yml matched exclusion entry: bad.yml
INFO     Executing syntax check on playbook good.yml (0.66s)
DEBUG    Examining good.yml of type playbook
WARNING  Unable to load module ansible.windows.win_feature at good.yml:6 for options validation
WARNING  Listing 1 violation(s) that are fatal
name[casing]: All names should start with an uppercase letter.
good.yml:2

DEBUG    Attempting to release lock 140189477433872 on /root/.cache/ansible-compat/5694d0/.lock
DEBUG    Lock 140189477433872 released on /root/.cache/ansible-compat/5694d0/.lock
Read documentation for instructions on how to ignore specific rule violations.

DEBUG    Determined rule-profile order: {'internal-error': (0, 'min'), 'load-failure': (1, 'min'), 'parser-error': (2, 'min'), 'syntax-check': (3, 'min'), 'command-instead-of-module': (4, 'basic'), 'command-instead-of-shell': (5, 'basic'), 'deprecated-bare-vars': (6, 'basic'), 'deprecated-local-action': (7, 'basic'), 'deprecated-module': (8, 'basic'), 'inline-env-var': (9, 'basic'), 'key-order': (10, 'basic'), 'literal-compare': (11, 'basic'), 'jinja': (12, 'basic'), 'no-free-form': (13, 'basic'), 'no-jinja-when': (14, 'basic'), 'no-tabs': (15, 'basic'), 'partial-become': (16, 'basic'), 'playbook-extension': (17, 'basic'), 'role-name': (18, 'basic'), 'schema': (19, 'basic'), 'name': (20, 'basic'), 'var-naming': (21, 'basic'), 'yaml': (22, 'basic'), 'name[template]': (23, 'moderate'), 'name[imperative]': (24, 'moderate'), 'name[casing]': (25, 'moderate'), 'spell-var-name': (26, 'moderate'), 'avoid-implicit': (27, 'safety'), 'latest': (28, 'safety'), 'package-latest': (29, 'safety'), 'risky-file-permissions': (30, 'safety'), 'risky-octal': (31, 'safety'), 'risky-shell-pipe': (32, 'safety'), 'galaxy': (33, 'shared'), 'ignore-errors': (34, 'shared'), 'layout': (35, 'shared'), 'meta-incorrect': (36, 'shared'), 'meta-no-tags': (37, 'shared'), 'meta-video-links': (38, 'shared'), 'meta-version': (39, 'shared'), 'meta-runtime': (40, 'shared'), 'no-changed-when': (41, 'shared'), 'no-changelog': (42, 'shared'), 'no-handler': (43, 'shared'), 'no-relative-paths': (44, 'shared'), 'max-block-depth': (45, 'shared'), 'max-tasks': (46, 'shared'), 'unsafe-loop': (47, 'shared'), 'avoid-dot-notation': (48, 'production'), 'sanity': (49, 'production'), 'fqcn': (50, 'production'), 'import-task-no-when': (51, 'production'), 'meta-no-dependencies': (52, 'production'), 'single-entry-point': (53, 'production'), 'use-loop': (54, 'production')}
              Rule Violation Summary
 count tag          profile  rule associated tags
     1 name[casing] moderate idiom

DEBUG    Registered VCS backend: bzr
DEBUG    Registered VCS backend: git
DEBUG    Registered VCS backend: hg
DEBUG    Registered VCS backend: svn
DEBUG    Found ansible-lint 6.14.6 dist
Failed after basic profile, 1/5 star rating: 1 failure(s), 0 warning(s) on 1 files.
A new release of ansible-lint is available: 6.14.6 → 6.16.2
```

**Comment by gone-for-coding** [comment]
Closing as a duplicate of https://github.com/ansible/ansible-lint/issues/3477
